### PR TITLE
[ESD-2250] Encode/decode functions in cpp message traits

### PR DESCRIPTION
**Pull request by woodfell**
While it was originally thought that there wouldn't be a need to deal with encoded messages using the V4 API the ephemeris swiftlet work introduced a need with how it dumps the see after startup. 

Add wrappers for the message type specific encode/decode/length functions to c++ message traits

### CODE REVIEW

**Review by RReichert — APPROVED** [review]
LGTM

**Review by samvrlewis — APPROVED** [review]

**Review by antrikshsrivastava — APPROVED** [review]

**Review by silverjam — APPROVED** [review]
Done with CI fixes, let's wait for CI to pass before merging (it should pass, but would like to double check)

**Pull request merged**
